exec/memory.go

```diff
@@ -208,7 +208,16 @@ func (vm *VM) currentMemory() {
 func (vm *VM) growMemory() {
 	_ = vm.fetchInt8() // reserved (https://github.com/WebAssembly/design/blob/27ac254c854994103c24834a994be16f74f54186/BinaryEncoding.md#memory-related-operators-described-here)
 	curLen := len(vm.memory) / wasmPageSize
-	n := vm.popInt32()
+	n := vm.popUint32()
+
+	maxMem := vm.module.Memory.Entries[0].Limits.Maximum
+	newPage := uint64(n + uint32(len(vm.memory)/wasmPageSize))
+
+	if newPage > 1<<16 || newPage > uint64(maxMem) {
```

Comment:
it would be great to have test for this.

Reply:
this bug is found by running the testcases in `testsuite/memory_grow.wast`. so have already covered this.